scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,74 @@ def visit_functiondef(self, node):
     # this line makes it work for async functions
     visit_asyncfunctiondef = visit_functiondef
 
+class CheckApiVersion(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-api-version"
+    priority = -1
+    msgs = {
+        "C4748": (
+            "Check api version",
+            "api-version-not-keyword",
+            "Acceot a keyword argument called api_version.",
+        ),
+    }
+    options = (
+        (
+            "ignore-api-version-incorrect",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow for no keyword api version.",
+            },
+        ),
+    )
+    ignore_clients = ["PipelineClient", "AsyncPipelineClient", "ARMPipelineClient", "AsyncARMPipelineClient"]
+
+    def __init__(self, linter=None):
+        super(CheckApiVersion, self).__init__(linter)             
+
+    def visit_classdef(self, node):
+        """Visits every class in file and checks if it is a client.
+        If it is a client, it checks that there is an api_version keyword.
+
+        :param node: class node
+        :type node: ast.ClassDef
+        :return: None
+        """
+
+        #Need to take into account python 2 versus version 3 of python
```

Comment:
if we're just looking at docstring I don't believe there should need to be py2/3 compat?